mps-sync-plugin/src/main/kotlin/org/modelix/mps/sync/plugin/ModelSyncService.kt

```diff
@@ -87,27 +90,33 @@ class ModelSyncService : Disposable {
         callback: (() -> Unit),
     ) {
         coroutineScope.launch {
-            logger.info { "disconnecting to server: ${modelClient.baseUrl}" }
-            syncService.disconnectModelServer(modelClient, callback)
-            logger.info { "disconnected server: ${modelClient.baseUrl}" }
+            try {
+                logger.info { "disconnecting to server: ${modelClient.baseUrl}" }
+                syncService.disconnectModelServer(modelClient)
```

Comment:
thought: `disconnectModelServer` might throw an execption in which case a client in undined state might be still in `activeClients`.

That is why in thet long-term I would like only to have one active client per project.

Reply:
fixed in [49ee4cd](https://github.com/modelix/modelix.mps-plugins/pull/73/commits/49ee4cdb9aeeb1cbd46bb798084dbdebe1cf7efa)

Reply:
If `disconnectModelServer`fails, `activeClients` would still contain `modelClient`.

Reply:
Yes, that's true. But the hope would be that we could disconnect it the second time we try. However, it might be a hopeless try.

What would be your expectation in this case? Shall we remove the client from the `connectionsModel`?

Reply:
`activeClients` does not exist anymore. I must have looked at some old code.

>  But the hope would be that we could disconnect it the second time we try.
Interesting solution.